webpack/webpack-embedded-dependencies-plugin/src/EmbeddedDependenciesWebpackPlugin.ts

```diff
@@ -0,0 +1,363 @@
+
+              packages.push({
+                name,
+                version,
+                license,
+                licenseSource,
+                copyright
+              });
+            } else {
+              // If there is no license file path, we still should populate the other required fields
+              const copyright: string | undefined = parsePackageAuthor(data);
+
+              packages.push({
+                name,
+                version,
+                license,
+                copyright
+              });
+            }
+          })
+            .catch((error: unknown) => {
+              this._emitWebpackError(compilation, 'Failed to process embedded dependencies', error);
+            })
+            .finally(() => {
+              // Sort the packages by name after we have processed all of them
+              // Note this is a mutable sort
+              Sort.sortBy(packages, (pkg) => pkg.name);
+            });
+
+          const dataToStringify: IEmbeddedDependenciesFile = {
+            embeddedDependencies: packages
+          };
+
+          compilation.emitAsset(this._outputFileName, new sources.RawSource(JSON.stringify(dataToStringify)));
+
+          if (this._generateLicenseFile) {
+            // We should try catch here because generator function can be output from user config
+            try {
+              compilation.emitAsset(
+                this._generatedLicenseFilename,
+                new sources.RawSource(this._generateLicenseFileFunction(packages))
+              );
+            } catch (error: unknown) {
+              this._emitWebpackError(compilation, 'Failed to generate license file', error);
+            }
+          }
+
+          return;
+        }
+      );
+    });
+  }
+
+  /**
+   * Default error handler for try/catch blocks in the plugin
+   * try/catches emit errors of type `unknown` and we need to handle them based on what
+   * type the error is. This function provides a convenient way to handle errors and then
+   * propagate them to webpack as WebpackError objects on `compilation.errors` array.
+   *
+   * @remarks
+   * _If we need to push errors to `compilation.warnings` array, we should just create a companion function
+   * that does the same thing but pushes to `compilation.warnings` array instead._
+   *
+   * @example
+   * ```typescript
+   * try {
+   *   // do some operation
+   *   FileSystem.readFile('some-file');
+   * } catch (error: unknown) {
+   *   this._emitWebpackError(compilation, 'Failed to do some operation', error);
+   * }
+   * ```
+   */
+  private _emitWebpackError(compilation: Compilation, errorMessage: string, error: unknown): void {
+    // If the error is a string, we can just emit it as is with message prefix and error message
+    if (typeof error === 'string') {
+      compilation.errors.push(new WebpackError(`${PLUGIN_ERROR_PREFIX}: ${errorMessage}: ${error}`));
```

Comment:
Deduplicate the `compilation.errors.push(` call by simply assigning a variable via a few different paths, then do the push once.